pkg/frontend/querymiddleware/codec_test.go

```diff
@@ -295,6 +305,9 @@ func TestResponseRoundtrip(t *testing.T) {
 		},
 	} {
 		t.Run(tc.name, func(t *testing.T) {
+			reg := prometheus.NewPedanticRegistry()
+			codec := NewPrometheusCodec(reg)
```

Comment:
(nit) why not use `newTestPrometheusCodec` here?

Reply:
`newTestPrometheusCodec()` doesn't expose the `Registry`, which is needed later on to check the metrics that are emitted by `prometheusCodec`.